test/base/MockFactory.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package base
+
+import com.codahale.metrics.Timer
+import connectors.CustomsDeclareExportsMovementsConnector
+import metrics.MovementsMetrics
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import org.scalatest.mockito.MockitoSugar
+import play.api.http.Status.INTERNAL_SERVER_ERROR
+import play.api.test.Helpers.NO_CONTENT
+import services.{CustomsCacheService, SubmissionService}
+import uk.gov.hmrc.http.HttpResponse
+import uk.gov.hmrc.http.cache.client.CacheMap
+
+import scala.concurrent.Future
+
+object MockFactory extends MockitoSugar {
+
+  private val emptyCacheMap = CacheMap("", Map.empty)
+
+  def buildCustomsCacheServiceMock: CustomsCacheService = {
+    val customsCacheServiceMock = mock[CustomsCacheService]
+    when(customsCacheServiceMock.fetchAndGetEntry(any(), any())(any(), any(), any()))
+      .thenReturn(Future.successful(None))
+    when(customsCacheServiceMock.fetch(any())(any(), any())).thenReturn(Future.successful(None))
+    when(customsCacheServiceMock.cache(any(), any(), any())(any(), any(), any()))
+      .thenReturn(Future.successful(emptyCacheMap))
+    when(customsCacheServiceMock.remove(any())(any(), any())).thenReturn(Future.successful(HttpResponse(NO_CONTENT)))
```

Comment:
And second comment. I think it's good to create mock definition for every case. Not some general object with everything inside. We can have object with all mocks etc, but this `when` etc. I think it's easier to have it inside specific tests. It's really difficult to reuse those things in other tests, because e.g. you need to have different behaviour or something like this and in practice you need to override this. In more general approach, combination of mocks etc is specific for test case. Thoughts after working with backend movements, it take some time to understand where this mock is, how it works and what mocks. Those stuff are used only in specs, so I think it's enough to create it for specific test cases and don't worry about duplications.

Reply:
I totally agree with you. The reason behind creating this `MockFactory` is not to provide mocks ready to use in every test case. It is there because Mockito does not work perfectly with Scala (so far) and in most cases the default return value is `null`. To avoid `null`s I decided to create `MockFactory`, to define default values that match return types.